File: treeView/src/main/java/org/sqlunet/treeview/control/Link.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>.
 */

package org.sqlunet.treeview.control;

/**
 * Link Data
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public abstract class Link
{
	/**
	 * Id used in link
	 */
	public final long id;

	/**
	 * Constructor
	 *
	 * @param id id
	 */
	protected Link(long id)
	{
		this.id = id;
	}

	/**
	 * Process
	 */
	abstract public void process();
}

File: treeView/src/main/java/org/sqlunet/treeview/control/Link.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.treeview.control

/**
 * Link Data
 *
 * @param id id
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
abstract class Link protected constructor(
    /**
     * Id used in link
     */
    @JvmField val id: Long,
) {

    /**
     * Process
     */
    abstract fun process()
}
